### PR TITLE
Remove relayer fee router

### PR DESCRIPTION
**Pull request by LayneHaber**
## Description

Contains changes from #1994 

- Removes relayer fee router
- Renames `relayerFeeRouter` -> `relayerFeeVault`
- Still stores relayer fees sent

## Type of change

- [ ] Docs change / dependency upgrade
- [ ] Configuration / tooling changes
- [ ] Refactoring
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires changes in customer code

## High-level change(s) description - from the user's perspective

- Removes relayer fee router in exchange for gelato 1Balance
- Still need addresses for testnet / mainnet for deployments

## Related Issue(s)

N/A.

### CODE REVIEW

File: packages/deployments/contracts/contracts/core/connext/facets/BridgeFacet.sol [hunk 1]
```diff
@@ -392,7 +389,9 @@ contract BridgeFacet is BaseConnextFacet {
   function bumpTransfer(bytes32 _transferId) external payable nonReentrant whenNotPaused {
     if (msg.value == 0) revert BridgeFacet__bumpTransfer_valueIsZero();
 
+    // TODO: should we store the fees or is there an easier way to reference whats been paid?
```
Comment (jakekidd) [inline packages/deployments/contracts/contracts/core/connext/facets/BridgeFacet.sol:392]: only 5k for update here, at least. idk if there's anything else we can do here w/o adding sec assumptions

File: packages/deployments/contracts/contracts/core/connext/facets/RelayerFacet.sol [hunk 2]
```diff
@@ -4,25 +4,23 @@ pragma solidity 0.8.15;
...
-   * @notice Emitted when the relayerFeeRouter variable is updated
-   * @param oldRouter - The relayerFeeRouter old value
-   * @param newRouter - The relayerFeeRouter new value
+   * @notice Emitted when the relayerFeeVault variable is updated
+   * @param oldRouter - The relayerFeeVault old value
+   * @param newRouter - The relayerFeeVault new value
    * @param caller - The account that called the function
    */
-  event RelayerFeeRouterUpdated(address oldRouter, address newRouter, address caller);
+  event RelayerFeeVaultUpdated(address oldRouter, address newRouter, address caller);
```
Comment (jakekidd) [inline packages/deployments/contracts/contracts/core/connext/facets/RelayerFacet.sol:23]: nit: update params

File: packages/deployments/contracts/contracts/core/connext/facets/BridgeFacet.sol (hunk 1 above)
Comment (jakekidd) [reply packages/deployments/contracts/contracts/core/connext/facets/BridgeFacet.sol:392]: maybe we should just track all the fees in the relayer fee vault if we're gonna transfer there anyway

File: packages/deployments/contracts/contracts/core/connext/facets/RoutersFacet.sol [hunk 3]
```diff
@@ -542,12 +545,21 @@ contract RoutersFacet is BaseConnextFacet {
...
 
     // Sanity check: router is approved.
     if (!_isRouterWhitelistRemoved() && !getRouterApproval(_router))
       revert RoutersFacet__addLiquidityForRouter_badRouter();
 
+    if (s.domain == canonical.domain) {
+      // Sanity check: caps not reached
+      uint256 custodied = IERC20(_local).balanceOf(address(this)) + _amount;
+      uint256 cap = s.caps[key];
+      if (custodied > cap && cap > 0) {
```
Comment (jakekidd) [inline packages/deployments/contracts/contracts/core/connext/facets/RoutersFacet.sol:558]: nit: reverse this conditional statement to be more logical
```
if (cap > 0 && custodied > cap)
```

Comment (jakekidd) [reply packages/deployments/contracts/contracts/core/connext/facets/RoutersFacet.sol:558]: no point in doing check if there is no cap

Comment (jakekidd) [reply packages/deployments/contracts/contracts/core/connext/facets/RoutersFacet.sol:558]: no cap

File: packages/deployments/contracts/contracts/core/connext/libraries/LibConnextStorage.sol [hunk 4]
```diff
@@ -119,10 +114,10 @@ struct AppStorage {
   // 1
   uint256 LIQUIDITY_FEE_NUMERATOR;
   /**
-   * @notice The local nomad relayer fee router.
+   * @notice The local address for relayer fees
```
Comment (jakekidd) [inline packages/deployments/contracts/contracts/core/connext/libraries/LibConnextStorage.sol:117]: nit: address for relayer fee *vault*

File: packages/deployments/contracts/contracts/core/connext/facets/BridgeFacet.sol [hunk 5]
```diff
@@ -495,6 +494,19 @@ contract BridgeFacet is BaseConnextFacet {
...
 
+        // Enforce liquidity caps
+        // NOTE: safe to do this before the swap because canonical domains do
+        // not hit the AMMs (local == canonical)
+        if (isCanonical) {
+          // NOTE: this method includes router liquidity as part of the caps,
+          // not only the minted amount
+          uint256 custodied = IERC20(local).balanceOf(address(this)) + _amount;
+          uint256 cap = s.caps[key];
+          if (custodied > cap && cap > 0) {
```
Comment (jakekidd) [inline packages/deployments/contracts/contracts/core/connext/facets/BridgeFacet.sol:505]: dup nit: reverse this conditional statement to be more logical
```
if (cap > 0 && custodied > cap)
```

File: packages/deployments/contracts/contracts/core/connext/facets/BridgeFacet.sol [hunk 6]
```diff
@@ -495,6 +494,19 @@ contract BridgeFacet is BaseConnextFacet {
         // Set boolean flag
         isCanonical = _params.originDomain == canonical.domain && local == _asset;
 
+        // Enforce liquidity caps
+        // NOTE: safe to do this before the swap because canonical domains do
+        // not hit the AMMs (local == canonical)
+        if (isCanonical) {
+          // NOTE: this method includes router liquidity as part of the caps,
+          // not only the minted amount
+          uint256 custodied = IERC20(local).balanceOf(address(this)) + _amount;
```
Comment (jakekidd) [inline packages/deployments/contracts/contracts/core/connext/facets/BridgeFacet.sol:503]: idk how i feel about the fact that the custodied funds are cumulative regardless of the toke

File: packages/deployments/contracts/contracts/core/connext/facets/BridgeFacet.sol (hunk 1 above)
Comment (LayneHaber) [reply packages/deployments/contracts/contracts/core/connext/facets/BridgeFacet.sol:392]: we can emit it in events and track it offchain, so ill remove

File: packages/deployments/contracts/contracts/core/connext/facets/BridgeFacet.sol [hunk 7]
```diff
@@ -391,11 +392,15 @@ contract BridgeFacet is BaseConnextFacet {
...
+    _bumpTransfer(_transferId);
+  }
 
-    s.relayerFees[_transferId] += msg.value;
+  function _bumpTransfer(bytes32 _transferId) internal {
+    if (msg.value == 0) revert BridgeFacet__bumpTransfer_valueIsZero();
+    Address.sendValue(payable(s.relayerFeeVault), msg.value);
 
-    emit TransferRelayerFeesUpdated(_transferId, s.relayerFees[_transferId], msg.sender);
+    // console.log("emitting event");
```
Comment (jakekidd) [inline packages/deployments/contracts/contracts/core/connext/facets/BridgeFacet.sol:402]: remove

File: packages/deployments/contracts/contracts/core/connext/facets/BridgeFacet.sol [hunk 8]
```diff
@@ -391,11 +392,15 @@ contract BridgeFacet is BaseConnextFacet {
    * @param _transferId - The unique identifier of the crosschain transaction
    */
   function bumpTransfer(bytes32 _transferId) external payable nonReentrant whenNotPaused {
-    if (msg.value == 0) revert BridgeFacet__bumpTransfer_valueIsZero();
+    _bumpTransfer(_transferId);
+  }
 
-    s.relayerFees[_transferId] += msg.value;
+  function _bumpTransfer(bytes32 _transferId) internal {
+    if (msg.value == 0) revert BridgeFacet__bumpTransfer_valueIsZero();
```
Comment (jakekidd) [inline packages/deployments/contracts/contracts/core/connext/facets/BridgeFacet.sol:399]: this check should just be in the `external` fn imo.

File: packages/deployments/contracts/contracts/core/connext/facets/BridgeFacet.sol [hunk 9]
```diff
@@ -640,16 +647,10 @@ contract BridgeFacet is BaseConnextFacet {
...
       // If there are no routers for this transfer, this `execute` must be a slow liquidity route; in which
       // case, we must make sure the transfer's been reconciled.
-      if (!reconciled) revert BridgeFacet__execute_notReconciled();
-    }
-
-    // Require that this transfer has not already been executed. If it were executed, the `transferRelayer`
-    // would have been set in the previous call (to enable the caller to claim relayer fees).
-    if (s.transferRelayer[transferId] != address(0)) {
-      revert BridgeFacet__execute_alreadyExecuted();
+      if (status != DestinationTransferStatus.Reconciled) revert BridgeFacet__execute_notReconciled();
```
Comment (jakekidd) [inline packages/deployments/contracts/contracts/core/connext/facets/BridgeFacet.sol:650]: might as well make this whole block an `else if`

File: packages/deployments/contracts/contracts/core/connext/facets/BridgeFacet.sol [hunk 10]
```diff
@@ -533,10 +538,12 @@ contract BridgeFacet is BaseConnextFacet {
       _params.nonce = s.nonce++;
     }
 
-    // Store the relayer fee.
+    // Handle the relayer fee.
```
Comment (jakekidd) [inline packages/deployments/contracts/contracts/core/connext/facets/BridgeFacet.sol:541]: Remove `NOTE` below: I think it's not relevant anymore, as it was referring to handling native assets as input assets

Comment (LayneHaber) [reply packages/deployments/contracts/contracts/core/connext/facets/BridgeFacet.sol:541]: note is still relevant for event purposes -- basically just saying that the bridged amount is what is hashed in the transfer id so you need to wait until after swap to make sure you have the right transfer

File: packages/deployments/contracts/contracts/core/connext/facets/BridgeFacet.sol (hunk 9 above)
Comment (LayneHaber) [reply packages/deployments/contracts/contracts/core/connext/facets/BridgeFacet.sol:650]: ehh would require me to handle the else

Comment (jakekidd) [reply packages/deployments/contracts/contracts/core/connext/facets/BridgeFacet.sol:649]: `else` would just `pass` - no decision there

but it's fine, nbd

**Review by jakekidd — APPROVED** [review]

File: packages/deployments/contracts/contracts/core/connext/facets/BridgeFacet.sol (hunk 9 above)
Comment (LayneHaber) [reply packages/deployments/contracts/contracts/core/connext/facets/BridgeFacet.sol:649]: no you need it to throw if the status fails, not just do nothing (indicating pass of sanity checks)

Comment (jakekidd) [reply packages/deployments/contracts/contracts/core/connext/facets/BridgeFacet.sol:649]: lol hold up. I think this
```solidity
if (pathLength != 0) {
  if (status != DestinationTransferStatus.None) revert BridgeFacet__execute_badFastLiquidityStatus();
} else if (status != DestinationTransferStatus.Reconciled) {
  revert BridgeFacet__execute_notReconciled();
}
```
is the equivalent of this
```solidity
if (pathLength != 0) {
  if (status != DestinationTransferStatus.None) revert BridgeFacet__execute_badFastLiquidityStatus();
} else {
  if (status != DestinationTransferStatus.Reconciled) revert BridgeFacet__execute_notReconciled();
}
```

In the first case, there is no `else` case needed. If the `else if` conditional is not met, then we just proceed.

**Pull request merged**